common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala

```diff
@@ -36,60 +38,17 @@ object StubAPI {
-                                           stub: Option[Stub]
-                                         ): Future[(Option[String], Option[String])] = {
-    stub
-      .flatMap(s => Some(s.plannedBookSectionId, s.plannedBookSectionId)) match {
-      case Some((Some(bId), Some(bsId))) => getPrintLocationDescriptionFromIds(contentAPI, bId, bsId)
-      case _ => Future.successful((None, None))
-    }}
-
-  private def getActualPrintLocationDescriptionFromExternalData(
-                                           contentAPI: ContentAPI,
-                                           externalData: Option[ExternalData]
-                                         ): Future[(Option[String], Option[String])] = {
-    externalData
-      .flatMap(e => Some(e.actualBookId, e.actualBookSectionId)) match {
-      case Some((Some(bId), Some(bsId))) => getPrintLocationDescriptionFromIds(contentAPI, bId, bsId)
-      case _ => Future.successful((None, None))
-    }}
-
-  private def getPrintLocationDescriptionFromIds(contentAPI: ContentAPI, bId: Long, bsId: Long): Future[(Option[String], Option[String])] =
-    {
-      for {
-        maybeBook <- contentAPI.getTagInternalName(bId)
-        maybeBookSection <- contentAPI.getTagInternalName(bsId)
-      } yield (maybeBook, maybeBookSection) match {
-        case (Some(book), Some(bookSection)) => (Some(book + " >> " + bookSection), Some(bookSection))
-        case ( _, Some(bookSection)) => (None, Some(bookSection))
-        case _ => (None, None)
-      }
-    }
-
-  def getStubByComposerId(contentAPI: ContentAPI, composerId: String): ApiResponseFt[Option[Stub]] =
+  def getStubByComposerId(stubDecorator: StubDecorator, composerId: String): ApiResponseFt[Option[Stub]] = {
     for {
       res <- ApiResponseFt.Async.Right(getRequest(s"content/$composerId"))
       json <- parseBody(res.body)
       maybeStub <- extractDataResponseOpt[Stub](json)
-      maybeExternalData = maybeStub.flatMap(_.externalData)
-      actualPrintLocationDescription <- ApiResponseFt.Async.Right(getActualPrintLocationDescriptionFromExternalData(contentAPI, maybeExternalData))
-      plannedPrintLocationDescription <- ApiResponseFt.Async.Right(getPlannedPrintLocationDescriptionFromStub(contentAPI, maybeStub))
-    } yield maybeStub.map(
-      stub => stub
-        .copy(
-          externalData = stub.externalData.map(
-            e => e.copy(
-              longActualPrintLocationDescription = actualPrintLocationDescription._1,
-              shortActualPrintLocationDescription = actualPrintLocationDescription._2)
-          )
-        )
-        .copy(
-          longPlannedPrintLocationDescription = plannedPrintLocationDescription._1,
-          shortPlannedPrintLocationDescription = plannedPrintLocationDescription._2
-        )
-      )
+      maybeDecoratedStub <- ApiResponseFt.Async.Right(maybeStub match {
+        case Some(stub) => stubDecorator.withPrintLocationDescriptions(stub).map(Some(_))
```

Comment:
`.map(Some(_))` is necessary to satisfy the type. It's not great and help wanted to improve it!

Reply:
```suggestion
ApiResponseFt.Async.Right(maybeStub.map(stub => stubDecorator.withPrintLocationDescriptions(stub)))
```

Reply:
😢  this doesn't satisfy the types.

```scala
ApiResponseFt.Async.Right(maybeStub.map(stub => stubDecorator.withPrintLocationDescriptions(stub)))
```

yields an `Option[Future[Stub]]`, but a `Future[Option[Stub]]` is needed.